src/woodbury_pd_mat.jl

```diff
@@ -90,4 +88,6 @@ end
 # NOTE: the parameterisation to scale up the Woodbury matrix is not unique. Here we
 # implement one way to scale it.
 *(a::WoodburyPDMat, c::Real) = WoodburyPDMat(a.A, a.D * c, a.S * c)
-*(c::Real, a::WoodburyPDMat) = a * c
\ No newline at end of file
+*(c::Real, a::WoodburyPDMat) = a * c
+*(c::Diagonal{T}, a::WoodburyPDMat) where {T<:Real} = c * Matrix(a)
```

Comment:
these two are not needed, right?
```julia
julia> A = randn(4, 2); # on master

julia> D = Diagonal(randn(2).^2 .+ 1);

julia> S = Diagonal(randn(4).^2 .+ 1);

julia> d = Diagonal(rand(4));

julia> W = WoodburyPDMat(A, D, S);

julia> d*W == d*Matrix(W)
true

julia> W*d == Matrix(W)*d
true
```

Reply:
Oh you are correct.

Reply:
Oh not in the case of the Real - just using the abstractmatrix fallback will cause it to densify.